docs/synapse/deploymentguide.rst

```diff
@@ -0,0 +1,355 @@
+
+The AHA service is used for service discovery and acts as a CA to issue host/user certificates used to link
+Synapse services. Other Synapse services will need to be able to resolve the IP address of the AHA service
+by name, so it is likely that you need to create a DNS A/AAAA record in your existing resolver. When you are
+using AHA, the only host that needs DNS or other external name resolution is the AHA service.
+
+For this example deployment, we will name our AHA server ``aha.loop.vertex.link`` and assume DNS records
+have been created to resolve the FQDN to an IPv4 / IPv6 address of the host running the container.
+
+Create the container directory::
+
+    mkdir -p /srv/syn/aha/storage
+
+Create the ``/srv/syn/aha/docker-compose.yaml`` file with contents::
+
+    version: "3.3"
+    services:
+      aha:
+        user: 999
+        image: vertexproject/synapse-aha:v2.x.x
+        network_mode: host
+        restart: unless-stopped
+        volumes:
+        - ./storage:/vertex/storage
+
+Create the ``/srv/syn/aha/storage/cell.yaml`` file with contents::
+
+    aha:name: aha
+    aha:network: loop.vertex.link
+    aha:urls: ssl://aha.loop.vertex.link
+    dmon:listen: ssl://aha.loop.vertex.link?ca=loop.vertex.link
+    provision:listen: ssl://aha.loop.vertex.link:27272/
+
+.. note::
+
+    Don't forget to replace ``loop.vertex.link`` with your chosen network name!
+
+Change ownership of the storage directory to the user you will use to run the container::
+
+    chown -R 999 /srv/syn/aha/storage
+
+Start the container using ``docker-compose``::
+
+    docker-compose -f /srv/syn/aha/docker-compose.yaml pull
+    docker-compose -f /srv/syn/aha/docker-compose.yaml up -d
+
+To view the container logs at any time you may run the following command on the *host* from the
+``/srv/syn/aha`` directory::
+
+    docker-compose logs -f
+
+You may also execute a shell inside the container using ``docker-compose`` from the ``/srv/syn/aha``
+directory on the *host*. This will be necessary for some of the additional provisioning steps::
+
+    docker-compose exec aha /bin/bash
+
+Deploy Axon Service
+===================
+
+In the Synapse service architecture, an Axon provides a place to store arbitrary bytes/files as binary
+blobs and exposes APIs for streaming files in and out regardless of their size.  Given sufficient file system
+size, an Axon can be used to efficiently store and retrieve very large files as well as a high number
+(easily billions) of files.
+
+**Inside the AHA container**
+
+Generate a one-time use provisioning URL::
+
+    python -m synapse.tools.aha.provision.service 00.axon
+
+You should see output that looks similar to this::
+
+    one-time use URL: ssl://aha.loop.vertex.link:27272/<guid>?certhash=<sha256>
+
+**On the Host**
+
+Create the container directory::
+
+    mkdir -p /srv/syn/axon/storage
+    chown -R 999 /srv/syn/axon/storage
+
+Create the ``/srv/syn/00.axon/docker-compose.yaml`` file with contents::
```

Comment:
In terms of just running through this on a single host, the HTTPS ports led to a collision.  Maybe disable those by default or provide a note here?